enterprise_gateway/services/kernels/remotemanager.py

```diff
@@ -571,8 +573,17 @@ def write_connection_file(self):
             self.hb_port = ports[3]
             self.control_port = ports[4]
 
-            return super(RemoteKernelManager, self).write_connection_file()
+            cf = super(RemoteKernelManager, self).write_connection_file()
+            username=self.user_overrides.get('KERNEL_USERNAME',None)
+            if username:
+                uid=getpwnam(username).pw_uid
+                guid=grp.getgrnam(username).gr_gid
+                os.chown(self.connection_file,uid,guid)
```

Comment:
Had to change ownership of connection files. Because connection files are created by the gateway process (root) and when the kernel is launched as a user `Permission denied` happens. Alternative approaches are welcome~